superset/config.py

```diff
@@ -986,6 +989,16 @@ def CSV_TO_HIVE_UPLOAD_DIRECTORY_FUNC(  # pylint: disable=invalid-name
 # See here: https://github.com/dropbox/PyHive/blob/8eb0aeab8ca300f3024655419b93dad926c1a351/pyhive/presto.py#L93  # pylint: disable=line-too-long,useless-suppression
 PRESTO_POLL_INTERVAL = int(timedelta(seconds=1).total_seconds())
 
+# Allow list of custom authentications for each DB engine.
+# Example:
+# ALLOWED_EXTRA_AUTHENTICATIONS: Dict[str, List[str]] = {
+#     "trino": [
+#         "your.module:AuthClass",
+#         "another.extra:auth_method",
+#     ],
+# }
```

Comment:
We could make this a bit more user friendly and save some work:

```python
from your.module import AuthClass
ALLOWED_EXTRA_AUTHENTICATIONS = {
    "trino": {
        "custom_auth": AuthClass,
        ...
    },
}
```

This way users can refer to the auth by a readable name like "custom_auth", and the module is already imported so you don't need to implement `import_string`.

Reply:
That is a good idea. I implement it now.